matlab/polar-traversal/polarCoordinateTraversalTest.m

```diff
@@ -524,3 +524,227 @@ function testSimulhits(testCase)
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% The center of the circle is on the fourth quadrant with positive x and y
+% direction
+function testFourthQuadrantPositiveXPositiveY(testCase)
+    min_bound = [-30.0, -30.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [10.0, -25.0];
+    ray_direction = [1.3, 5.9];
+    circle_center = [15.0, -15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 30.0;
+   
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,2,3,3,2,1];
+        expected_thetaVoxels = [2,2,2,1,1,1];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% The center of the circle is on the fourth quadrant with negative x and y
+% direction
+function testFourthQuadrantNegativeXNegativeY(testCase)
+    min_bound = [-30.0, -30.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [26.0, -7.0];
+    ray_direction = [-1.3, -1.2];
+    circle_center = [15.0, -15.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 30.0;
+   
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,2,3,3,3,2,1];
+        expected_thetaVoxels = [0,0,0,3,2,2,2];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% The center of the circle is on the origin of the grid with positive x and
+% y direction
+function testOriginPositiveXPositiveY(testCase)
+    min_bound = [-30.0, -30.0];
+    max_bound = [30.0, 30.0];
+    ray_origin = [-10.0, -7.0];
+    ray_direction = [1.3, 3.0];
+    circle_center = [0.0, 0.0];
+    circle_max_radius = 10.0;
+    num_radial_sections = 3;
+    num_angular_sections = 4;
+    t_begin = 0.0;
+    t_end = 30.0;
+   
+    verbose = false;
+    
+    [rVoxels, thetaVoxels] = polarCoordinateTraversal(min_bound, max_bound, ray_origin, ray_direction, ...
+        circle_center, circle_max_radius, num_radial_sections, num_angular_sections, t_begin, t_end, verbose);
+        expected_rVoxels     = [1,1,2,1];
+        expected_thetaVoxels = [2,1,1,1];
+        
+        verifyEqual(testCase, rVoxels, expected_rVoxels);
+        verifyEqual(testCase, thetaVoxels, expected_thetaVoxels);
+end
+
+% The center of the circle is on the origin of the grid with negative x and
```

Comment:
This test fails. Here is what I get:
```
rVoxels =

     1     1     2     1


thetaVoxels =

     2     1     1     1
```
Furthermore, the ray direction is positive in the x- and y-directions. I'm assuming this is a typographical error somewhere?

Reply:
I just updated the fixed version of NegativeXNegativeY test